src/conn/conn_log.c

```diff
@@ -133,8 +133,13 @@ __logmgr_config(
 		FLD_SET(conn->log_flags, WT_CONN_LOG_RECOVER_ERR);
 
 	WT_RET(__wt_config_gets(session, cfg, "log.zero_fill", &cval));
-	if (cval.val != 0)
+	if (cval.val != 0) {
+		if (F_ISSET(conn, WT_CONN_READONLY))
```

Comment:
Some configs get overwritten (in the override function) and some, if specified, are conflicting and get an error.  In particular, those that ask for specific writing operations like zero-fill does, is in conflict and gives an error rather than silently changing the value.  Also particularly those where the conflict is caused by explicit request since it is not the default.